turtle_graphics/turtle_graphics.js

```diff
@@ -0,0 +1,143 @@
+class Turtle {
+    constructor(x, y) {
+        this.x = x;
+        this.y = y;
+        this.direction = 0;
+
+        if (this.x === undefined) {
```

Comment:
another way you could have done the same would be
1. `constructor (x = 0, y = y)` or
2. `this.x = x || 0 this.y = y || 0`